docs: restructure flow yaml instructions (#4574)
jina-ai/jina: docs/fundamentals/flow/create-flow.md

Creating a Flow, on its face, means instantiating a Python object.
More importantly, however, creating and configuring a Flow means defining your {ref}`search microservice architecture <architecture-overview>`.

> It defines how your Executors are tied together, and how your data, well, *flows*.

Reviewer: ```suggestion
It defines how your Executors are connected together and how your data *flows* through them.
```

The pun is cute, but might be distracting

Every Flow can be defined either purely in Python, or be loaded from a YAML file.

````{admonition} Hint
:class: hint

> For professional use we recommend YAML files to configure your Flows.

Reviewer 2: ```suggestion
For production use we recommend YAML files to configure your Flows.
```

This is because YAML files are:

- independent of Python source code
- easy to edit, maintain and extend

> - human-readable

Reviewer: Probably meant that you don't need to understand the Python code surrounding the definition, but just read the YAML object definitions as their own things.

````

## Instantiate a Flow

The most trivial Flow is the empty Flow and, like any other Flow, it can be instantiated purely in Python, or from a
YAML file:

> ````{tab} Pythonic style

Reviewer: ```suggestion
````{tab} Python
```

> ````{tab} Pythonic style

Reviewer 2: Do we keep Pythonic style ? Or just say Python ?

```
````

> `````{tab} Load from YAML

Reviewer: ```suggestion
`````{tab} YAML
```

with f:  # Using it as a Context Manager will start the Flow
    f.post(on='/search')  # This sends a request to the /search endpoint of the Flow
```

````{admonition} Hint: Dump Flow configuration
:class: hint

> In addition to loading a Flow from a YAML file, you can also dump an existing Flow to YAML. To do so, execute `f.save_config('path/to/flow.yml')`.

Reviewer: ```suggestion
In addition to loading a Flow from a YAML file, you can also save an existing Flow configuration to YAML. To do so, execute `f.save_config('path/to/flow.yml')`.
```

I'd say the word "dump" should be avoided a bit, if we can
